### PR TITLE
Continue executing tests if irreplaceable security manager is installed

### PR DESCRIPTION
**Pull request by Aaron1011**
Previously, TestWorker failed to handle a SecurityException
thrown when it called System.setSecurityManager(null). This
commit makes TestWorker properly handle any SecurityException that is
thrown, allowing tests that install an irreplaceable security manager to
exit cleanly.

Fixes issue #5305

### CODE REVIEW

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy [hunk 1]
```diff
@@ -501,4 +501,39 @@ class TestingIntegrationTest extends JUnitMultiVersionIntegrationSpec {
...
+                            }
+                            if (perm.getName().equals("setSecurityManager")) {
+                                throw new SecurityException("You cannot replace this security manager!");
+                            }
+                        }
+                    });
+                }
+            }
+        """
+        expect:
```
Comment (marcphilipp) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy]: Can you please add a blank line above this line?

File: subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestWorker.java [hunk 2]
```diff
@@ -73,8 +73,13 @@ public void execute(final WorkerProcessContext workerProcessContext) {
...
         } finally {
             LOGGER.info("{} finished executing tests.", workerProcessContext.getDisplayName());
-            // Clean out any security manager the tests might have installed
-            System.setSecurityManager(null);
+
+            try {
+                // Clean out any security manager the tests might have installed
+                System.setSecurityManager(null);
+            } catch (SecurityException e) {
+                LOGGER.info("Unable to clear out SecurityManager, continuing anyway.", e);
```
Comment (marcphilipp) [inline subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestWorker.java]: I think we should check whether the `SecurityManager` was changed by the test and only attempt to reset it if it was. If the test did install a `SecurityManager`, we should log a warning "Unable to clear out SecurityManager installed by a test, continuing anyway".

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy [hunk 3]
```diff
@@ -501,4 +501,39 @@ class TestingIntegrationTest extends JUnitMultiVersionIntegrationSpec {
...
+
+    @Issue("https://github.com/gradle/gradle/issues/5305")
+    def "test can install an irreplaceable SecurityManager"() {
+        given:
+        buildFile << """
+            apply plugin:'java'
+            ${mavenCentralRepository()}
+            dependencies { testCompile 'junit:junit:4.12' }
+        """
+        and:
```
Comment (marcphilipp) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy]: Please add a blank line above and remove the one below.

**Comment by Aaron1011** [comment]
@marcphilipp: I've made the changes you requested.

**Comment by Aaron1011** [comment]
Apparently, some of the tests are failing (according to the status checks). However, I can't reproduce the issue locally, and all of the 'Details' links require me to login.

**Comment by marcphilipp** [comment]
Apparently, there was a problem downloading dependencies from Bintray. I've restarted the CI build. Sorry for the delay!

**Comment by Aaron1011** [comment]
@marcphilipp: Thanks, that did it!

**Comment by Aaron1011** [comment]
@marcphilipp The 'Ci Status' check was passing earlier today, but it's failing now.

**Comment by Aaron1011** [comment]
All of the checks seem to be passing again.

**Comment by Aaron1011** [comment]
@marcphilipp Should I be concerned about the 'Branch Build Accept' failure?

**Comment by marcphilipp** [comment]
Nope, it was caused by a somewhat flaky performance test. It turned green after a rerun.

**Comment by marcphilipp** [comment]
Thanks, @Aaron1011! 👍

**Pull request merged**